common/app/navigation/Navigation.scala

```diff
@@ -0,0 +1,145 @@
+package navigation
+
+import common.{Edition, Navigation, editions}
+import _root_.model.Page
+import NavLinks._
+
+import scala.annotation.tailrec
+
+sealed trait NavNode {
+  def children: Seq[NavLink]
+}
+
+case class NavLink(
+  title: String,
+  url: String,
+  longTitle: String = "",
+  // TODO: get rid of below
+  iconName: String = "",
+  children: Seq[NavLink] = Nil
+)
+
+
+case class NavRoot private(val children: Seq[NavLink]) extends NavNode {
+
+  def findDescendantByUrl(url: String): Option[NavLink] = {
+
+    @tailrec
+    def find(children: Seq[NavLink]): Option[NavLink] = {
+      children match {
+        case Nil => None
+        case head :: tail if (head.url == url) => Some(head)
+        case head :: tail => find(tail ++ head.children)
+      }
+    }
+    find(children)
+  }
+
+  def findParentByUrl(maybeCurrentNavLink: Option[NavLink]): Option[NavLink] = {
+
+    maybeCurrentNavLink.map(currentNavLink => {
```

Comment:
I suggest you leave that out for better composability; this method should not have to deal with all the ceremony of unfolding an option and providing an alternative. It is just a changing a `findParentByUrl(maybeLink)` to `maybeLink.flatMap(findParentByUrl)`.